### PR TITLE
Control group with validation

### PR DESCRIPTION
**Pull request by andrzejkupczyk**
Simplifies the validation of form fields. Makes use of the session store implementation.

### CODE REVIEW

File: tests/spec/Bootstrapper/FormSpec.php [hunk 1]
```diff
@@ -195,6 +195,12 @@ function it_overrides_the_submit_method()
         );
     }
 
+    function it_can_get_validation_errors()
+    {
+        $this->hasErrors('foo')->shouldBeBool();
+        $this->getFormattedError('foo')->shouldBeString();
+    }
+
 }
```
Comment (PatrickRose) [inline tests/spec/Bootstrapper/FormSpec.php]: Please could you add a test for the hasErrors method?

File: src/Bootstrapper/ControlGroup.php [hunk 2]
```diff
@@ -166,6 +166,21 @@ public function withHelp($help)
     }
 
     /**
+     * Adds validation error if occured
+     *
+     * @param  string $name
+     * @return $this
+     */
+    public function withValidation($name)
```
Comment (PatrickRose) [inline src/Bootstrapper/ControlGroup.php]: Please could you add a test for this?

**Comment by PatrickRose** [comment]
I like this a lot. Minor comments added

**Comment by andrzejkupczyk** [comment]
Great, I'm glad that these changes are worth taking into account. Writing tests is not my forte :stuck_out_tongue_winking_eye: I had problems with mocking session store although I will try again later.

**Comment by PatrickRose** [comment]
If you get stuck feel free to send me an email!

**Comment by andrzejkupczyk** [comment]
Thanks Patrick but It was a point of honor with me to write these tests alone ;)

File: src/Bootstrapper/ControlGroup.php (hunk 2 above)
Comment (andrzejkupczyk) [inline src/Bootstrapper/ControlGroup.php]: Done.

File: tests/spec/Bootstrapper/FormSpec.php (hunk 1 above)
Comment (andrzejkupczyk) [inline tests/spec/Bootstrapper/FormSpec.php]: Done.

**Comment by PatrickRose** [comment]
Sorry it's taken me so long, merging in!

**Comment by andrzejkupczyk** [comment]
No problem. Thank you, Patrick.

**Pull request merged**